include/deal.II/base/types.h

```diff
@@ -41,6 +41,12 @@ namespace types
    */
   typedef unsigned int subdomain_id;
 
+  /**
+   * The type used for global indices of vertices.
+   */
+  typedef unsigned long long int global_vertex_index;
+#  define DEAL_II_VERTEX_INDEX_MPI_TYPE MPI_UNSIGNED_LONG_LONG
```

Comment:
I'm actually a big fan of introducing typedefs with different names. If you see a variable
```
  std::vector<types::global_vertex_index> tmp;
```
you know exactly what's stored in it. I think that's far more self-documenting than if you had reused an existing name.